### DISCUSSION TITLE
Ruby stuck on 2.6 version

### DISCUSSION THREAD

**Discussion by HenryAugustiano** [rbenv/ruby-build#2380] [Build failures]
Hi! I'm new to Ruby, I am trying to update the ruby version but somehow it keeps using the old version instead of the newly installed version. I can't find any woking solutions online

```
username@HenMacbook-Pro ~ % brew upgrade ruby

Warning: ruby 3.3.1 already installed
username@HenMacbook-Pro ~ % ruby -v

ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.x86_64-darwin23]
```

**Comment by HenryAugustiano** [comment]
also tried using `rbenv install 3.3.1` 

output:
==> Installed ruby-3.3.1 to /Users/username/.rbenv/versions/3.3.1

NOTE: to activate this Ruby version as the new default, run: rbenv global 3.3.1
username@HenMacbook-Pro ~ % rbenv global 3.3.1
username@HenMacbook-Pro ~ % ruby -v
ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.x86_64-darwin23]

**Comment by mislav** [comment]
Hi! You're doing almost everything right, and you're likely just one small step away from succeeding. Let's set you on the right track.

First of all, there is a system version of Ruby at `/usr/bin/ruby` that is at version 2.6.10. Because there is already a system version of Ruby, doing `brew install ruby` or `brew upgrade ruby` does install a newer Ruby version but **does not place it in your PATH** (Homebrew calls this "keg-only" formula). That means that just typing `ruby -v` in your shell will still output 2.6.10, because the PATH still contains only `/usr/bin/ruby`. You can find what Homebrew installed for you by doing ``echo `brew --prefix ruby`/bin``. For more information on this, you can view the output of `brew info ruby`.

Now, if you want to use a newer version of Ruby with rbenv, you're on the right track to, but I suspect a similar PATH problem. I see you installed Ruby 3.3.1 and set it as rbenv "global" version. This is all good. But `ruby -v` still prints 2.6.10, which tells me that rbenv isn't properly set up in your PATH, which is likely due to the `eval "$(rbenv init - zsh)"` line not being in your `~/.zshrc`. Make sure that that line is there and restart your terminal. Then, `which ruby` should show `~/.rbenv/shims/ruby` instead of `/usr/bin/ruby`, and your rbenv global Ruby version should start to take effect.

**Reply by jBoscW** [reply]
is there any other command besides `eval "$(rbenv init - zsh)" ` that should be in the zsh file? 

I tried deleting the ones that were there related to ruby (very dumb in hindsight I know) but nothing changed when I input `ruby -v`. 

Inputting `which ruby` does output `/Users/myuser/.rbenv/shims/ruby` though

[RESOLVED] `ruby -v` now outputs the latest version after 
1) `brew reinstall ruby`
 2) after the steps above, and
3) with anaconda deactivated (not sure if that one mattered)
